app/code/Magento/Sitemap/Model/ResourceModel/Catalog/Product.php

```diff
@@ -326,7 +322,7 @@ public function getCollection($storeId)
         )->joinLeft(
             ['url_rewrite' => $this->getTable('url_rewrite')],
             'e.entity_id = url_rewrite.entity_id AND url_rewrite.is_autogenerated = 1'
-            . $urlRewriteMetaDataCondition
+            . ' AND url_rewrite.target_path NOT LIKE "%/category/%"'
```

Comment:
That might kill the performance. 
Isn't there any other way to get rid of these URLs?